### PR TITLE
Fix windows compilation for Pfem dependent applications and add Pfem to appveyor

### PR DESCRIPTION
**Pull request by josep-m-carbonell**
Some error fixes and changes in the Solid, Contact, Constitutive, Delaunay and Pfem applications. Adding KRATOS_API for windows compilation.

PfemFluid only som resizes for a future Amatrix compilation.

In the core only the configure file related to Appveyor

### CODE REVIEW

**Review by jginternational — DISMISSED** [review]
I think that 43k additions and 14k deletions in 363 files, and 87 commits are too much to check.
So I hope this fixes the Windows compilation. This is why I think that smaller pull request but more frequently is a better policy

**Comment by josep-m-carbonell** [comment]
You are right, I use to check it with git difftool, there are no changes in the core but, there are some changes in the ContactMechanics FRICTION_LAW that I could not extract from the rest of the mixed improvements.

File: scripts/build/appveyor/configure_appveyor_VS2017.bat [hunk 1]
```diff
@@ -13,6 +13,8 @@ cmake .. -G "Visual Studio 15 2017 Win64"
 -DMESH_MOVING_APPLICATION=ON                                                    ^
 -DSOLID_MECHANICS_APPLICATION=ON                                                ^
 -DCONSTITUTIVE_MODELS_APPLICATION=ON                                            ^
+-DADD_DEPENDENT_APPLICATIONS=ON                                                 ^
```
Comment (maceligueta) [inline scripts/build/appveyor/configure_appveyor_VS2017.bat]: @josep-m-carbonell  I like the concept of ADD_DEPENDENT_APPLICATIONS. However, in the case of this specific file, it is good to know which apps are being compiled. I've checked this file many times, and I know that @roigcarlo did as well. And ADD_DEPENDENT_APPLICATIONS is something that hides the total number of applications being compiled. IMO it is cool for the regular use in our own configures, but too opaque here...

Comment (josep-m-carbonell) [reply scripts/build/appveyor/configure_appveyor_VS2017.bat]: Ok, its better to show the applications here. I will commit the correction.

**Comment by josep-m-carbonell** [comment]
Finally, I have set the apps of PFEM as OFF to appveyor.

**Comment by maceligueta** [comment]
Why don't you want to add them to Appveyor?

**Comment by josep-m-carbonell** [comment]
It does not compiles because tetgen is not checkout in this repositori. Maybe in the future with a local copy of qhull we can activate it.

**Comment by maceligueta** [comment]
We could set Appveyor to download this submodule. Maybe @roigcarlo can help on this.

**Review by jginternational — APPROVED** [review]

**Comment by maceligueta** [comment]
@roigcarlo . Can we try to download submodules from Appveyor?

**Comment by maceligueta** [comment]
We would need the Appveyor script to include the following.
```
-DDELAUNAY_MESHING_APPLICATION=ON                                               ^
-DCONTACT_MECHANICS_APPLICATION=ON                                              ^
-DPFEM_APPLICATION=ON     ^
-DPFEM_SOLID_MECHANICS_APPLICATION=ON     ^
-DPFEM_FLUID_DYNAMICS_APPLICATION=ON     ^
```

please confirm @josep-m-carbonell  and @AFranci that there are no missing ONs.

**Comment by roigcarlo** [comment]
Testing in #3081

**Pull request merged**